### ISSUE TITLE
Websocket error after update iOS 10.3.1

### ISSUE THREAD

**Issue by Anopetia** [MicrosoftDX/Vorlonjs#401]
Hi,

Everything was fine since yesterday.
We upgrade our IOS device to version 10.3.1 and now we got "Websocket error" from vorlon.

We can see  the console.log from the device on our cordova app, but are not enable for example to send command (like alert or whatever), and we cannot reach the device information anymore.

When the device's launch we got this (see screenshot).

![image](https://cloud.githubusercontent.com/assets/7024320/24840816/9ca49770-1d42-11e7-9707-e18291a99ccd.png)

Any idea? Everything work fine on Android.

**Comment by Anopetia** [comment]
Uninstall; Reinstall the app,and the problem's gone.

Weird.

**Comment by Anopetia** [comment]
Speaking too fast, probem came back.... Exactly the same.... Just after reload the app.

**Comment by Anopetia** [comment]
AdBlock, Ghostery, seems to be the problem.

Disable all chrome plugin and no problem anymore....

**Comment by Anopetia** [comment]
Deactivating DomExplorer, And ObjectExplorer seems to remove the error.
Interactive console is working fine, but we cannot explorer dom and object.

**Comment by phil-lgr** [comment]
I can reproduce the same error my iPhone and also my mac:

Here is the stack on Safari 10.1 

![image](https://cloud.githubusercontent.com/assets/9644867/24941371/f6d3bdd0-1f17-11e7-86f6-165f04651626.png)
![image](https://cloud.githubusercontent.com/assets/9644867/24941389/0b505142-1f18-11e7-961a-3aabc9d31ae8.png)
![image](https://cloud.githubusercontent.com/assets/9644867/24941426/6e75f2c2-1f18-11e7-927d-16370cb15d01.png)

When I put a break point here:

![image](https://cloud.githubusercontent.com/assets/9644867/24941518/2bde2f50-1f19-11e7-9156-f458837fb415.png)

I see that it's throwing in json3 library.

Let us know how we can help to troubleshoot the issue

**Comment by meulta** [comment]
Hey guys!

Thanks for reporting this, I will have a look today :)

**Comment by Anopetia** [comment]
Cool Merci :)

**Comment by meulta** [comment]
Ok so I can reproduce it but it seems there is a racing condition here as it does not happen everytime.
I am going to love fixing this... :) 
Keep you posted as soon as I know more

**Comment by Anopetia** [comment]
Yes, it's like random, I cannot reproduce anytime too, but most part of the time.

**Comment by meulta** [comment]
So i digged a little and apparently there is a regression in Safari in this build around websocket.
Someone already created a bug issue in the webkit bugzilla: https://bugs.webkit.org/show_bug.cgi?id=170463 

I will give them until next week to fix it and if they don't I will be forced to implement a workaround in Vorlon. I do not want to do it if I can avoid it.

Does it sound ok to you?

**Comment by Anopetia** [comment]
Yep, for now we don't use dom explorer or object explorer.

Calling console.log(object) or  console.log(DOMHtml), do the job anyway, even if it's less user friendly.

Thank you for your attention meulta.

Waiting for the fix so.

**Comment by michaelfaisst** [comment]
@meulta Any update on this allready? :)

**Comment by natterstefan** [comment]
@michaelfaisst good question. :)

**Comment by antonfire** [comment]
Gonna test with iOS 10.3.3 beta to see if this is resolved, will report back

**Comment by hossein-nasr** [comment]
I still face this problem in the last Vorlon js and Safari version. any update?

**Comment by jf-patenaude** [comment]
I was searching for this same issue in a different context and I think it's fixed in 10.3.3.   I couldn't reproduce the bug with 10.3.3
